.rhdh/scripts/install-registry-proxy-images.sh

```diff
@@ -16,164 +16,185 @@
-        echo "[ERROR] Failed to copy $quay_image. Exiting."
-        exit 1
-    }
+# Clean up only the temporary directories you created
+for IMAGE in "${IMAGES[@]}"; do
+    rm -rf /tmp/"${IMAGE##*/}"
 done
 
-# Step 4: Apply ImageContentSourcePolicy
-echo "Creating ImageContentSourcePolicy..."
-oc apply -f - <<EOF
-apiVersion: operator.openshift.io/v1alpha1
-kind: ImageContentSourcePolicy
-metadata:
-  name: quay-registry
-spec:
-  repositoryDigestMirrors:
-  - mirrors:
-    - ${REMOTE_REGISTRY}
-    source: quay.io
-EOF
+# Create a temporary directory for YAMLs
+TMPDIR=$(mktemp -d)
+trap 'rm -fr "$TMPDIR"' EXIT
 
-# Step 5: Create CatalogSource in OpenShift
-echo "Creating CatalogSource..."
-oc apply -f - <<EOF
+# Set up CatalogSource for the IIB image
+infof "[INFO] Creating CatalogSource..."
+cat <<EOF > "$TMPDIR/catalogsource.yml"
 apiVersion: operators.coreos.com/v1alpha1
 kind: CatalogSource
 metadata:
-  name: $CATALOG_NAME
-  namespace: $NAMESPACE
+  name: rhdh-catalogsource
+  namespace: ${NAMESPACE_CATALOGSOURCE}
 spec:
   sourceType: grpc
-  image: $LOCAL_REGISTRY/$NAMESPACE/$(basename $IIB_IMAGE)
-  displayName: Custom Catalog Source
-  publisher: Red Hat
-  updateStrategy:
-    registryPoll:
-      interval: 15m
+  image: ${UPSTREAM_IIB}
+  publisher: IIB testing Red Hat Developer Hub
+  displayName: IIB testing catalog Red Hat Developer Hub
 EOF
 
-# Step 6: Create OperatorGroup and Subscription
-if ! oc get namespace "$OPERATOR_NAMESPACE" > /dev/null 2>&1; then
-    oc create namespace "$OPERATOR_NAMESPACE"
-fi
+oc apply -f "$TMPDIR/catalogsource.yml"
 
-echo "Creating OperatorGroup..."
-oc apply -f - <<EOF
+# Optional: Install the operator if specified
+if [[ -n "$TO_INSTALL" ]]; then
+    infof "[INFO] Installing Operator: ${TO_INSTALL}"
+
+    # Create OperatorGroup
+    cat <<EOF > "$TMPDIR/operatorgroup.yml"
 apiVersion: operators.coreos.com/v1
 kind: OperatorGroup
 metadata:
-  name: $OPERATOR_NAME-operator-group
-  namespace: $OPERATOR_NAMESPACE
-spec:
-  targetNamespaces:
-  - $OPERATOR_NAMESPACE
+  name: ${TO_INSTALL}-operatorgroup
+  namespace: ${NAMESPACE_SUBSCRIPTION}
 EOF
+    oc apply -f "$TMPDIR/operatorgroup.yml"
 
-echo "Creating Subscription for the RHDH 1.3 CI Operator..."
-oc apply -f - <<EOF
+    # Create Subscription
+    cat <<EOF > "$TMPDIR/subscription.yml"
 apiVersion: operators.coreos.com/v1alpha1
 kind: Subscription
 metadata:
-  name: $OPERATOR_NAME
-  namespace: $OPERATOR_NAMESPACE
+  name: ${TO_INSTALL}
+  namespace: ${NAMESPACE_SUBSCRIPTION}
 spec:
-  channel: $OLM_CHANNEL
+  channel: ${OLM_CHANNEL}
+  name: ${TO_INSTALL}
+  source: rhdh-catalogsource
+  sourceNamespace: ${NAMESPACE_CATALOGSOURCE}
   installPlanApproval: Automatic
-  name: $OPERATOR_NAME
-  source: $CATALOG_NAME
-  sourceNamespace: $NAMESPACE
 EOF
+    oc apply -f "$TMPDIR/subscription.yml"
+fi
 
-echo "RHDH 1.3 CI Operator has been deployed from the cached images."
+infof "[INFO] Completed successfully!"
```

Comment:
at this point you should have:
* loaded images from the CSV into the cluster
* created a subscription from the IIB/CSV

so the next step is to log in and create a RHDH Backstage instance from the subscription. Maybe we can tell the user that, and bonus points, provide a link to their 'installed operators' view at https://console-openshift-console.apps.<cluster-goes-here>/k8s/ns/rhdh-operator/operators.coreos.com~v1alpha1~ClusterServiceVersion?cluster-service-version=developer+hub ?